src/main/java/com/authservice/authservice/config/JwtTokenUtil.java

```diff
@@ -45,13 +48,75 @@ public String generateToken(User userDetails) {
      * @param subject
      * @return String
      */
-    private String doGenerateToken(Map<String, Object> claims, String subject, String id) {
+    private String doGenerateToken(Map<String, Object> claims, String subject) {
         log.info("Entering JwtTokenUtil.doGenerateToken with parameters claims {}, subject {}, {}", claims, subject, secret);
         return Jwts.builder().setSubject(subject).setIssuedAt(new Date(System.currentTimeMillis()))
                 .setExpiration(new Date(System.currentTimeMillis() + JWT_TOKEN_VALIDITY_MILLIS))
-                .setId(id)
                 .addClaims(claims)
                 .setIssuer(jwtIssuerName)
                 .signWith(SignatureAlgorithm.HS512, secret).compact();
     }
+
+    /**
+     * This function retrieve username from jwt token
+     *
+     * @param token
+     * @return String
+     */
+    public String getUsernameFromToken(String token) {
+        return getClaimFromToken(token, Claims::getSubject);
+    }
+
+    /**
+     * This function retrieve issuer name from the JWT token
+     *
+     * @param token
+     * @return
+     */
+    public String getIssuerFromToken(String token) {
+        return getClaimFromToken(token, Claims::getIssuer);
+    }
+
+    /**
+     * This function retrieve expiration date from jwt token
+     *
+     * @param token
+     * @return String
+     */
+    public Date getExpirationDateFromToken(String token) {
+        return getClaimFromToken(token, Claims::getExpiration);
+    }
+
+    public <T> T getClaimFromToken(String token, Function<Claims, T> claimsResolver) {
+        final Claims claims = getAllClaimsFromToken(token);
+        return claimsResolver.apply(claims);
+    }
+
+    /**
+     * This function is retrieving any information from token we will need the secret key
+     *
+     * @param token
+     * @return Claims
+     */
+    private Claims getAllClaimsFromToken(String token) {
+        return Jwts.parser().setSigningKey(secret).parseClaimsJws(token).getBody();
+    }
+
+    /**
+     * This function check if the token has expired
+     *
+     * @param token
+     * @return Boolean
+     */
+    private Boolean isTokenExpired(String token) {
+        final Date expiration = getExpirationDateFromToken(token);
+        return expiration.before(new Date());
+    }
+
+    public Boolean validateToken(String token, UserDetails userDetails) {
```

Comment:
put java docs on top of this as well. This is the most important function

Reply:
Done.